libs/execution_context/ebpf_native.c

```diff
@@ -1482,11 +1480,19 @@ _ebpf_native_load_programs(_Inout_ ebpf_native_module_instance_t* instance)
         if (result != EBPF_SUCCESS) {
             goto Done;
         }
+        native_program->loaded = true;
     }
 
 Done:
     if (result != EBPF_SUCCESS) {
-        _ebpf_native_clean_up_programs(instance->programs, instance->program_count, true);
+        // Release reference for each program.
```

Comment:
Since this block is used elsewhere, can you make this a macro or an inline function?